text/0000-qa-team.md

```diff
@@ -0,0 +1,106 @@
+- Title: qa-team
+- Authors: [joltz](mailto:joltz@protonmail.com)
+- Start date: Feb 12 2020
+- RFC PR: Edit if merged: [mimblewimble/grin-rfcs#0000](https://github.com/mimblewimble/grin-rfcs/pull/0000)
+- Tracking issue: [Edit if merged with link to tracking github issue]
+
+---
+
+# Summary
+[summary]: #summary
+
+* The QA team for Grin:
+  * works to ensure that a high standard of quality for contributions is upheld across the project
+  * develops tools and processes to help monitor and enforce the quality of work
+  * exists to make sure that changes to the project are quality, not to produce the quality changes themselves
+  * ensures that contributors are encouraged, supported and empowered to produce the best quality work to be integrated into Grin
+
+# Motivation
+[motivation]: #motivation
+
+Over time, with many people contributing to many different areas of Grin, some things can get lost in the mix including quality. It could be beneficial to have a team supported by members, processes and tools to help provide a consistent level of quality contributions.
+
+* Ensure that contributions to Grin are high quality
+* Empower contributors to make high quality contributions to Grin
+* Provide feedback from a QA perspective to micro and macro changes in Grin
+
+# Community-level explanation
+[community-level-explanation]: #community-level-explanation
+
+The QA team observes micro and macro changes occurring over time with various contributions to Grin. They help review PRs, have opinions on both development and governance decisions and provide tooling frameworks for contributors to use and maintain to ensure a consistent high level of quality of work across the Grin project. The QA team provides support for contributors to encourage them to continue to submit quality work.
+
+The day to day impact is more resources to empower contributors to make successful contributions to Grin by performing quality assurance testing and review on their own work. The long term impact is more consistency in the level of quality work that makes it into the Grin project over time.
+
+## Example
+
+In this example a large PR is submitted that makes a nontrivial change to the Grin node.
```

Comment:
I'm wondering to what extent non-developer members of the QA team will be expected to fully understand the implications of major changes, particularly to the point where they're expected to help review PRs.  Is it more a case of the QA team ensuring that tests exist or at least a testing strategy exists for verifying changes in PRs rather than the entire substance of the change?

Reply:
Thanks, this point is important to discuss as test coverage can be a bit subjective and the inability to fully understand the changes could result in QA approving coverage of a feature that isn't properly tested. My initial reaction is that the QA team is there to stand up processes, protocols and tools and ensure that they are properly used by contributors. As indicated, development knowledge is required to assess whether the test coverage is accurate (not to mention long term implications arising from the relevant changes that are not realizable with testing frameworks alone). 

We may decide that it is sufficient for a QA team to ensure that changes are trivially covered by the testing framework and that other tests don't fail. We may decide that there is no point in a QA team if it cannot make a more meaningful assessment than a binary test pass/fail, covered/not. It could be helpful to distinguish what is required for a minimally viable QA team that adds tangible value and what is required for an ideal high-performing QA team.